src/annotator/config/index.ts

```diff
@@ -31,12 +31,21 @@ type ConfigDefinitionMap = Record<string, ConfigDefinition>;
  */
 type Context = 'sidebar' | 'notebook' | 'profile' | 'annotator' | 'all';
 
+function throwInvalidContext(context: string): never {
+  throw new Error(`Invalid application context used: "${context}"`);
+}
```

Comment:
The `context` argument to `configurationKeys` always comes from us, right? It might be sufficient to just rely on TypeScript to validate that `configurationKeys` is always called with a legal `Context` value.

Reply:
Yep, I thought about that as well. There's room for improvement on type safety here. We could even try to define it in a way that the return type is more specific than `string[]` based on the value of the argument.

I will try to tackle this separately as soon as we have some room.